src/functional/iterator/transforms/unroll_reduce.py

```diff
@@ -1,14 +1,35 @@
+from itertools import groupby
+from typing import Iterable
+
 from eve import NodeTranslator
 from eve.utils import UIDs
 from functional.iterator import ir
 
 
+def all_equal(it: Iterable):
+    g = groupby(it)
+    return next(g, True) and not next(g, False)
```

Comment:
I would comment that this code comes from a trustworthy source (itertools recipe) as it is not really well readable on its own.